Add factory entropy injection to the example

Patater opened the pull request:
Add a fake factory entropy injection example and instructions in the README.

Also update the application configuration to explicitly enable PSA to enable the example to run on more targets. The example will not be very interesting if PSA is not enabled...

Patater committed:
Demonstrate entropy injection

Make the example demonstrate entropy injection by adding a new function,
called before psa_crypto_init(), that attempts to inject some fake
entropy into the system. Print a warning message if injecting entropy
fails.

Patater committed:
Enable PSA so that this example can run

This example requires PSA to be enabled in order for the PSA APIs to be
available for use. This example uses PSA Crypto APIs and thus requires
the target to have the PSA label in order to make those PSA APIs
available.

netanelgonen commented on main.cpp:
```diff
@@ -331,8 +332,33 @@ static void cipher_examples(void)
+    /* mbedtls_psa_inject_entropy() depends on both MBEDTLS_ENTROPY_NV_SEED and
+     * MBEDTLS_PSA_HAS_ITS_IO being enabled by the Mbed TLS configuration
+     * system. */
+#if defined(MBEDTLS_ENTROPY_NV_SEED) && defined(MBEDTLS_PSA_HAS_ITS_IO)
```
This may be wrong in case of 2 cores.
for example the Psoc6:
the M4 core will not have NV_SEED (so this function will not compile) but defined but the M0 will have it. (This comment applies on both of the defined )

MarceloSalazar replied:
Agreed. However, we may want to comment this is a generic implementation for single core systems and may need to be implemented on cases-by case basis?

Patater replied:
Updated

netanelgonen commented on mbed_app.json:
```diff
@@ -1,7 +1,8 @@
         "*": {
-             "platform.stdio-convert-newlines": true
+             "platform.stdio-convert-newlines": true,
+             "target.extra_labels_add": ["PSA"]
```
currently, in addition to PSA label, you must add the MBEDTLS_PSA_CRYPTO_C.

Patater replied:
Added via new Mbed TLS user configuration file

itayzafrir commented on main.cpp:
```diff
@@ -331,8 +332,33 @@ static void cipher_examples(void)
+        seed[i] = i;
+    }
+
+    int status =  mbedtls_psa_inject_entropy(seed, sizeof(seed));
```
Extra space after `=` sign
```suggestion
    int status = mbedtls_psa_inject_entropy(seed, sizeof(seed));
```

Patater replied:
Fixed

itayzafrir commented on main.cpp:
```diff
@@ -331,8 +332,33 @@ static void cipher_examples(void)
+    if (status) {
+        /* The device may already have an NV Seed injected, or another error
+         * may have happened during injection. */
+        mbedtls_printf("warning - this attempt at entropy injection failed\n");
```
Maybe print the error code here?

Patater replied:
Added

itayzafrir reviewed (changes requested):
A couple of minors.

MarceloSalazar commented:
Looks good.
Hope we can merged this to the mbed-os-5.11.0-oob branch to let us start the review asap.

Patater committed:
Enable PSA Crypto APIs by default

Since this example is not useful without PSA Crypto APIs, enable PSA
Crypto APIs by default via an Mbed TLS user configuration file, along
with a default implmentation of the PSA Entropy Injection API that
uses PSA storage (ITS).

itayzafrir reviewed (approved):
LGTM

Patater commented:
Updated so that this example should work whether being built for the NSPE of a partitioned system, or a single v7-M target.

netanelgonen reviewed (approved):
LGTM

itayzafrir reviewed (approved):
LGTM
